Automattic/pocket-casts-android
# Push PC data to Engage SDK

## Description

This PR integrates pushes data from Pocket Casts to the Engage SDK. I also added analytics handing.

## Testing Instructions

1. Use `debug` or `debugProd` variant.
2. Open the app and sign in.
3. Minimize the app.
4. Go through the first 4 steps from [the integration workflow](https://developer.android.com/guide/playcore/engage/workflow). It should contain our real data.
5. Tap on different things in the verification app and check that these analytics events have `engage_sdk` sources.

Events: `episode_detail_shown`, `podcast_screen_shown`, `incoming_share_list_shown`.
Sources: `engage_sdk_continuation`, `engage_sdk_featured`, `engage_sdk_recommendations`.

## Checklist
 
#### I have tested any UI changes...
